classes/rootfs_cpio.yaml

```diff
@@ -0,0 +1,27 @@
+inherit: [rootfs]
+
+depends:
+    - tools:
+        target-toolchain: host-compat-toolchain
+      use: [tools]
+      depends:
+          - utils::cpio
+
+buildToolsWeak: [cpio]
+buildSetup: |
+    # $1: rootfs source directory
+    # $2: output name
+    rootfsCpioCreateImage()
+    {
+        cat >create_cpio_img.sh <<EOF
+    $CREATE_SH_FILE
+
+        cd $1
+        find . \( -type d -name .debug -prune \) -o -print | \
```

Comment:
Pruning of any unwanted files need to be done in a previous step. Thats why my example is using $PWD/install as source directory. Actually this .debug thing here is wrong and I will remove it.

And yes, it is the idea that packages can add their own users (and maybe more stuff in the future). For example ntp or openssh are doing this (soonish). I talked with Jan about this and we came to the conclusion having a dedicated .bob directory for this kind of info is the best place. I agree that existing root recipes have to be adapted. However, because this is a directory where we can store any kind of additional info, this have to be done only once (hopefully).

And for the classes I added here: As usual with bob, you are free to use them (or only some of them) or not to use them at all and do your own stuff (like ignoring all those extra info).

Anyhow, I uploaded it to start a discussion, because I truly believe anybody needs this infrastructure at some point in one way or another. So having a standard way of doing things seems, to me, to be good.

Reply:
Hmm..we have all this in a separate config repo + some tools around. This way the packages stay pure and we have a better control about the stuff going into rootfs.

Maybe you can also add helper functions to add the user / devicetable files in the packages as well as a basement build-var to disable this?

Reply:
Yes, this would ofc be possible. However, just ignoring the .bob directories would also do that job, no?

My main sync loop looks like this:
```
mkdir -p install
for i in "${@}" ; do
         rsync -auHK --exclude ".debug/" --exclude ".git" --exclude ".bob" \
             $i/ install/
done
```

In my opinion the packages itself should describe their requirements somehow. In this way adding a new package to a root recipe will just magically do the right thing. Having this in extra packages means not just adding the new package, but also additional files somewhere else. I understand that there are multiple ways in doing things and we should ofc aim in supporting all of them.

Reply:
And another comment so you maybe understand better what we are doing: We are not building just one rootfs. We are building multiple rootfs's in one go. Usually with one base set of packages and then different packages on top for every other rootfs. Having the packages itself describe there requirements, makes this just easier to handle.

Reply:
I think there are multiple use cases here that we have to unpack.

At the heart of it, using packages should _just work_. So in the best case, there is some standard way to create a rootfs and it should not be much harder than using a class, declaring some dependencies and maybe add another file or two. Very much what buildroot provides today. OTOH this _must not_ preclude the possibility to do it differently. So if the system configuration, its users and whatnot are checked into a dedicated repository, then this has to work too.

The way how this might work is through the `.bob` directory. There a package can describe its required system configuration. Very much like the meta data that is included in a deb or rpm package. This will be picked up by these standard classes. If you have a completely separate configuration, all you have to do is ignore the `.bob` directory. Sure, this is something new but having reserved a full directory should make it easier to extend or ignore in the future.

I don't think it makes sense to control the creation of the `.bob` meta data in each package itself. Like the `.debug` directories where we also made a global decision. Having a `.bob` meta directory feels like the easiest option. Any by "reserving" a whole directory, it should be enough for the future.